compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -194,10 +194,13 @@ private void compileModulesInternal() {
         // add dependency manifest diagnostics
         diagnostics.addAll(packageContext().dependencyManifest().diagnostics().allDiagnostics);
         // add compilation diagnostics
-        for (ModuleContext moduleContext : packageResolution.topologicallySortedModuleList()) {
-            moduleContext.compile(compilerContext);
-            for (Diagnostic diagnostic : moduleContext.diagnostics()) {
-                diagnostics.add(new PackageDiagnostic(diagnostic, moduleContext.descriptor(), moduleContext.project()));
+        if (!packageResolution.diagnosticResult().hasErrors()) {
```

Comment:
There can be different types of errors added in the package resolution phase. E.g. cyclic dependencies, incompatible dependency version conflicts, and with IDL imports, there can be errors related to the client generation as well. Are we going to stop compilation for all these scenarios? This will stop giving suggestions in LS

Reply:
if it is only the cyclic dependency case that we need to handle, then we can use `packageResolution.dependencyGraph().findCycles()` which will return a Set

Reply:
We can't do this because of https://github.com/ballerina-platform/ballerina-lang/issues/37257

Reply:
Ack